geode-core/src/main/java/org/apache/geode/internal/cache/DiskInitFile.java

```diff
@@ -1310,7 +1311,16 @@ private void writeIFRecord(ByteBuffer bb) throws IOException {
   private void writeIFRecord(ByteBuffer bb, boolean doStats) throws IOException {
     assert lock.isHeldByCurrentThread();
     if (closed) {
-      throw new DiskAccessException("The disk store is closed", parent);
+      parent.getCache().getCancelCriterion().checkCancelInProgress();
```

Comment:
Could you extract this whole "if" block into a method "checkClosed"? We currently would call checkClosed from two places so this would get rid of some code duplication